actionpack/lib/action_controller/metal/redirecting.rb

```diff
@@ -92,10 +98,21 @@ def redirect_back(fallback_location:, allow_other_host: true, **args)
     #
     # All other options that can be passed to #redirect_to are accepted as
     # options and the behavior is identical.
-    def redirect_back_or_to(fallback_location, allow_other_host: true, **args)
-      referer = request.headers["Referer"]
-      redirect_to_referer = referer && (allow_other_host || _url_host_allowed?(referer))
-      redirect_to redirect_to_referer ? referer : fallback_location, **args
+    def redirect_back_or_to(fallback_location, allow_other_host: _allow_open_redirects, **options)
+      referer = request.referer || fallback_location
+      fallback_location = _allow_open_redirects ? fallback_location : nil
+
+      redirect_to referer, fallback_location: fallback_location, allow_other_host: allow_other_host, **options
+    end
+
+    def _compute_safe_redirect_to_location(request, options, response_options)
+      location = _compute_redirect_to_location(request, options)
+      if response_options[:allow_other_host] || _url_host_allowed?(location)
+        location
+      else
+        fallback_location = response_options[:fallback_location] || _raise_open_redirect(location)
+        _compute_redirect_to_location(request, fallback_location)
+      end
```

Comment:
```suggestion
      location = _compute_redirect_to_location(request, options)

      return location if response_options[:allow_other_host] || _url_host_allowed?(location)

      fallback_location = response_options[:fallback_location] || _raise_open_redirect(location)
      _compute_redirect_to_location(request, fallback_location)
```

does it work like that? just for curiosity.

Reply:
Yeah, that's equivalent. But since we don't want `redirect_to` to fallback, the function is now a little more simple.